lib/multierror/multierror.go

```diff
@@ -61,26 +77,22 @@ func (me MultiError) As(target interface{}) bool {
 }
 
 // Is for MultiError returns true if any of the errors listed can be considered of the target type.
-func (me MultiError) Is(target error) bool {
-	for _, err := range me {
+func (multi MultiError) Is(target error) bool {
+	for _, err := range multi {
 		if errors.Is(err, target) {
 			return true
 		}
 	}
 	return false
 }
 
-func (me MultiError) Error() string {
-	if len(me) == 1 && me[0] != nil {
-		return me[0].Error()
-	}
-
-	messages := []string{}
-	for _, err := range me {
+func (multi MultiError) Error() string {
+	var messages []string
+	for _, err := range multi {
 		if err == nil {
 			continue
 		}
 		messages = append(messages, err.Error())
 	}
-	return "Multiple errors:\n  " + strings.Join(messages, "\n  ")
+	return strings.Join(messages, Separator)
```

Comment:
"Multiple errors:" is dropped?

Reply:
I'd prefer to keep it, makes for a more human friendly and readable output.
(assuming we keep "Multiple errors:", please keep the special casing for a single error - again, goal is user friendliness / easier to understand output)

Reply:
I'm generally allergic to  uneditable formating  inside errors themselves. 
It would be much more idiomatic if instead we have another function that handles the error printing themselves. 
Something like this 

`errhandle.PrintErr(err)` <= and inside this we can check if it's a multierror or not and do error handling things. 

Ideally no one should just fmt.Printtln(err.Error())

Reply:
I suspect this change also makes it easier to test in the manner done in this PR - though if the test suggestions are followed (test the final error contains substrings for each included error, instead of joining the test errors manually and testing for equality) then this change can be dropped.

I think it's open for debate, but definitely not the most high-priority debate we have today, and there's no need for this to stall an otherwise good change that adds needed unit tests.

Reply:
uhm .. this regresses on error messages commonly shown by enkit login / enkit tunnel etc.
It goes from a clear message showing that multiple errors happened, to a naked array.

Pretty much all errors in go are full of hardcoded strings. eg, just grep the source code for fmt.Errorf of base libraries?

Reply:
Also, the recommended way to print an error is `%v` or `%w` in `fmt.Errorf`. I've almost never seen go code special casing an error for the purpose of printing.

Reply:
This is more of a molehill to die on than a hill, so I'll just revert it.

Reply:
But I want to point out that having multiple errors is NOT and error, which is why I wanted to remove it.